src/simcore_service_deployment_agent/auto_deploy_task.py

```diff
@@ -126,17 +126,18 @@ def add_prefix_to_services(
 async def generate_stack_file(
     app_config: dict[str, Any], git_task: GitUrlWatcher
 ) -> Path:
+    stack_recipe_cfg = app_config["main"]["docker_stack_recipe"]
+
     # collect repos informations
-    git_repos = {}
-    git_repos.update({x.repo_id: x for x in git_task.watched_repos})
+    git_repos: dict[RepoID, GitRepo] = {r.repo_id: r for r in git_task.watched_repos}
 
-    stack_recipe_cfg = app_config["main"]["docker_stack_recipe"]
     # collect files in one location
     dest_dir = stack_recipe_cfg["workdir"]
     if dest_dir == "temp":
         # create a temp folder
-        directoryName = tempfile.mkdtemp()
-        dest_dir = copy.deepcopy(directoryName)
+        directory_name = tempfile.mkdtemp()  # TODO: use asyncexitstack and aiofiles
```

Comment:
todo... missing or this is in 2/2?

Reply:
I refactored a similar one in  ``git_url_watcher`` using  ``aiofiles.tempdir.TemporaryDirectory`` and ``AsyncExitStack`` . I leave thisone as exercise for @mrnicegyu11  ;-)